pybind_instructions.md

```diff
@@ -0,0 +1,82 @@
+# Porting an algorithm from C++ to Python
+
+## Table of Contents
+- [Why?](#why)
+- [How it works](#how-it-works)
+- [Requirements](#requirements)
+
+## Why?
+The goal of this project is to allow users to take an algorithm written in C++ and translate it into runnable Python code.
+This could allow a user to develop C++ code and run it on the EmotiBit's firmware, software, and through Python, hitting three different metrics.
+
+## How it works
+- We use pybind11 to create bindings for existing c++ code to a dynamic python library. These bindings get added to `bindings.cpp`
+- These bindings essentially define the class/function prototypes, in the pybind11 format. An example of a binding would look like
+```c++
+PYBIND11_MODULE(EmotiBitPacket, m) {
+	py::class_<EmotiBitPacket>(m, "EmotiBitPacket")
+		.def(py::init<>());
+	m.def("createHeader", &EmotiBitPacket::createHeader);
+	m.def("headerToString", &EmotiBitPacket::headerToString);
+	//m.def("createPacket", &EmotiBitPacket::createPacket); 
+	py::class_<EmotiBitPacket::Header>(m, "Header")
+		.def(py::init<>());
+}
+```
+- The pybind11 python library (installed in the virtual environment) performs the mapping from c++ file to a `.pyd` file.
+- The flow to go from `.cpp` to `.pyd` looks like
+  - Create a `CMakeLists.txt` that lists the pybind module and its dependency on the source file library.
+  - Run `cmake` to create build files
+  - Build the files created in the previous step to create the `.pyd` file
+
+## Requirements
+In order to configure an algorithm for pybind, a new directory name `pybind` should be created under the root directory. This directory will hold all files related to the C++ to Python process.
+
+In addition, [EmotiBit ArduinoFilters](https://github.com/EmotiBit/EmotiBit_ArduinoFilters) must be present in the Arduino libraries folder (from library manager or manual download) so that the ArduinoMock library can be found by CMake.
```

Comment:
This is very specific to algos that required ArduinoFilters.
What if that is not a dependency?